autogtp/Production.cpp

```diff
@@ -21,6 +21,7 @@
 #include <QDir>
 #include <QFileInfo>
 #include <QThread>
+#include <QMutex>
```

Comment:
I included QMutex in each file that uses the name (which seems to be what you do) instead of relying on <file>.h or worse some other class to include it.

Reply:
Relying on the .h that matches the .cpp is fine and probably better style.

Reply:
updated to have rely on includes in .h